### PR TITLE
Adding Subcellular Localization (SCL) splits

### PR DESCRIPTION
**Pull request by joaquimgomez**
Once applied, this PR will add Subcellular Localization (SCL) splits.

**Proposed splits**:
- From the original Stärk et al. paper data:
    - `mixed_soft`: deepLoc train + deepLoc validation + deepLoc test
    - `mixed_hard`: deepLoc train + deepLoc validation + testHARD test
- From the original Thumuluri et al. DeepLoc 2.0 paper data:
    - `mixed_vs_human_2`: deepLoc 2 train (mixed, multi-label) + deepLoc 2 validation (mixed, multi-label) + deepLoc 2 test (human, multi-label)
- New proposals:
    - `human_soft`: deepLoc train + deepLoc validation + deepLoc test (only human proteins)
    - `human_hard`: deepLoc train + deepLoc validation + testHARD test (only human proteins)
    - `balanced`: same as `mixed_hard` but with the test set classes balanced with respect to the training set

Open issues:

- [x] How to carry out the balance of classes? Options:

1. Oversample the smallest classes: discarded due to the complexity of creating a large bunch of artificial sequences.
2. Undersample the biggest dataset: the best choice?
3. Are there other options?

- [x] Is it ok to choose the human sequences only by comparing the name of the proteins?

### CODE REVIEW

**Comment by sacdallago** [comment]
> Open issues:
> 
> * [ ]  How to carry out the balance of classes? Options:
> 
> 1. Oversample the smallest classes: discarded due to the complexity of creating a large bunch of artificial sequences.
> 2. Undersample the biggest dataset: the best choice?
> 3. Are there other options?

I think 2 is good enough. What is your interpretation of class balance here?
1. everything has equal weight, or 
2. that test + train follow the same distribution?

I think that 1. might result in a very challaging set, because the smallest class is very small. The goal of this split would be to assess whether some class has an easier signal than another to learn, I assume.

For 2., this should be easier (I think deeploc already has the same distribution in train and test, but not deeploc vs setHARD). Here the goal would probably be to see how much the model learns to predict distributions of classes regardless of the underlying signal.

> 
> * [ ]  Is it ok to choose the human sequences only by comparing the name of the proteins?

Here you mean, to compare their accessions to UniProt? Sure. You can also use [the ID mapping](https://www.uniprot.org/uploadlists/) to retrieve a list of proteins, and you can download the result as a CSV with the spieces/organism number. Human is 9606

**Comment by sacdallago** [comment]
Had a look at the notebook, looks good. A few points:
- `.cvs` vs `.csv`
- It would be good to have some plots of distributions for each split (like in [the other notebooks](https://github.com/J-SNACKKB/FLIP/blob/777ff8f63335b1dcf2eaaeb25043c7c245374c6d/collect_splits/3_gb1.ipynb):
  - Sequences in train vs test
  - Distribution of sequences per class in train vs. test

You may also want to add the splits ZIPped to this PR, together with an explanation README like [this one](https://github.com/J-SNACKKB/FLIP/tree/main/splits/gb1#dataset-description). Since this is the first classification problem added, you could add how many classes there are in this problem, etc. :)

**Comment by joaquimgomez** [comment]
@sacdallago Splits ready. I have finished splits, including the `balanced` dataset from your suggestions. I have also included the plots, the splits ZIPped and the README.

Regarding the DeepLoc 2.0 dataset with multi-labels, I have added it as we discussed in the meeting. I have a separate notebook with DeepLoc 2.0 split but with only one label per sequence, not included in this PR. If you consider that it could be good to have it, I can add it

File: splits/scl/README.md
Comment (sacdallago) [inline splits/scl/README.md:3]:
```suggestion
The six SCL (SubCellularLocation) splits stem from a [2021 publication](https://academic.oup.com/bioinformaticsadvances/article/1/1/vbab035/6432029) (based on a [2017 publication](https://academic.oup.com/bioinformatics/article/33/21/3387/3931857)) and a [2022 publicaiton](https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkac278/6576357) which aim at predicting protein subcelluar location.
```

File: splits/scl/README.md
Comment (sacdallago) [inline splits/scl/README.md:5]:
```suggestion
The possible subcellular localizations (in the splits, assigned to `TARGET`) are `Cytoplasm`, `Nucleus`, `Cell membrane`, `Mitochondrion`, `Endoplasmic reticulum`, `Lysosome/Vacuole`, `Golgi apparatus`, `Peroxisome`, `Extracellular` and `Plastid`.
```

File: splits/scl/README.md
Comment (sacdallago) [inline splits/scl/README.md:11]:
```suggestion
The `Swissprot_Train_Validation_dataset.csv` and the `hpa_testset.csv` are the training/validation and test set from the [DeepLoc v2 2022 publicaiton](https://academic.oup.com/nar/advance-article/doi/10.1093/nar/gkac278/6576357) used to create the DeepLoc 2.0 split.
```

File: splits/scl/README.md
Comment (sacdallago) [inline splits/scl/README.md:9]:
```suggestion
The `deeploc_our_train_set.fasta`, `deeploc_our_val_set.fasta`, `deeploc_our_test_set.fasta` and `setHARD.fast` are the source dataset from [the Light Attention (LA) 2021 publication](https://academic.oup.com/bioinformaticsadvances/article/1/1/vbab035/6432029) used to create the DeepLoc 1.0 splits. The `human_sequences.tsv` is used to recover the human proteins from the DeepLoc 1.0 set.
```

File: splits/scl/README.md
Comment (sacdallago) [inline splits/scl/README.md:17]:
```suggestion
All splits are classification splits as explained prior.
```

File: splits/scl/README.md
Comment (sacdallago) [inline splits/scl/README.md:39]: Please also add the DeepLoc v1 citation :)

**Review by sacdallago — CHANGES_REQUESTED** [review]
Looks good with a couple of comments. Once addressed, feel free to merge.

File: splits/scl/README.md
Comment (joaquimgomez) [reply splits/scl/README.md:17]: Oops, sorry, I copied an example to follow the structure and I didn't realize that I got out of this sentence.

**Comment by joaquimgomez** [comment]
@sacdallago Changes done. I can't merge since I have no write access to the repository.

**Review by sacdallago — APPROVED** [review]

**Pull request merged**
